nbdime/args.py

```diff
@@ -17,6 +19,62 @@ def __call__(self, parser, namespace, values, option_string=None):
         init_logging(level=level)
         set_nbdime_log_level(level)
 
+
+class IgnorableAction(argparse.Action):
+    """Adds the supplied positive options and negative/ignore version as well"""
+
+    def __init__(self, option_strings, dest, default=None, required=False, help=None):
+        opts = []
+        for opt in option_strings:
+            if len(opt) == 2 and opt[0] == '-':
+                if not opt[1].islower():
+                    raise ValueError('Single character flags should be lower-case for IgnorableAction')
+                opts.append(opt)
+                opts.append(opt.upper())
+            elif opt[:2] == '--':
+                opts.append(opt)
+                opts.append('--ignore-' + opt[2:])
+            else:
+                ValueError('Could not turn option "%s" into an IgnoreAction option.')
+
+        # Put positives first, negatives last:
+        opts = opts[0::2] + opts[1::2]
+
+        super(IgnorableAction, self).__init__(
+            opts, dest, nargs=0, const=None,
+            default=default, required=required,
+            help=help)
+
+    def __call__(self, parser, ns, values, option_string):
+        if len(option_string) == 2:
+            setattr(ns, self.dest, option_string[1].islower())
+        else:
+            setattr(ns, self.dest, option_string[2:2 + len('ignore')] != 'ignore')
+
+
+def process_exclusive_ignorables(ns, arg_names, default=True):
+    """Parse a set of ignorables.
+
+    It checks that all specified options are either all positive or all negative.
+    It then returns a namespace with the parsed options.
+    """
+    state = getattr(ns, arg_names[0])
+    for name in arg_names[1:]:
+        opt = getattr(ns, name)
+        if state is None:
+            state = opt
+        elif state != opt and opt is not None:
+            message = 'Arguments must either all be negative or all positive: %r' % (arg_names,)
+            raise argparse.ArgumentError(None, message)
+    if state is None:
+        state = not default
+
+    # Set all unset to the opposite of state (if all specified are True, others are False and vice versa)
+    for name in arg_names:
+        if getattr(ns, name) is None:
+            setattr(ns, name, not state)
```

Comment:
It's a little confusing that this is `not not default` in the default case, though I get why. It might be useful to use a more descriptive name for a boolean that came from user args than `state` such as `set_value` or `toggle`.

Maybe it would be a bit clearer above to have:

```python
if toggle is not None:
    # a value was specified, default unspecified items to the inverse
    default = not state
...
    setattr(ns, name, default)